### PR TITLE
Define new parameters in Search Tree configuration

### PR DESCRIPTION
**Pull request by terriervik**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

### CODE REVIEW

File: ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/process/search_tree/Tree.java [hunk 1]
```diff
@@ -86,6 +102,33 @@ private Tree() {
         return CompletableFuture.completedFuture(buildOutput(rootLeaf, optimalLeaf));
     }
 
+    /**
+     * Stop criterion check: maximum research depth reached
+     */
+    static boolean goDeeper(int currentDepth, int maxDepth) {
```
Comment (baptsegu) [inline ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/process/search_tree/Tree.java:108]: private

File: ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/process/search_tree/Tree.java [hunk 2]
```diff
@@ -86,6 +102,33 @@ private Tree() {
...
+        return currentDepth < maxDepth;
+    }
+
+    /**
+     * Stop criterion check: the remedial action has enough impact on the cost
+     */
+    static boolean improvedEnough(double oldCost, double newCost, double relativeImpact, double absoluteImpact) {
```
Comment (baptsegu) [inline ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/process/search_tree/Tree.java:115]: private

File: ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/process/search_tree/Tree.java [hunk 3]
```diff
@@ -86,6 +102,33 @@ private Tree() {
...
+        return oldCost - absoluteImpact > newCost && (1 - Math.signum(oldCost) * relativeImpact) * oldCost > newCost;
+    }
+
+    /**
+     * Stop criterion check: is positive or maximum margin reached?
+     */
+    static boolean doNewIteration(SearchTreeRaoParameters.StopCriterion stopCriterion, boolean hasImproved, double optimalCost) {
```
Comment (baptsegu) [inline ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/process/search_tree/Tree.java:122]: private

File: ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/process/search_tree/Tree.java [hunk 4]
```diff
@@ -56,8 +72,7 @@ private Tree() {
         Leaf optimalLeaf = rootLeaf;
         boolean hasImproved = true;
 
-        //TODO: generalize to handle different stop criterion
-        while (optimalLeaf.getCost(crac) > 0 && hasImproved) {
+        while (goDeeper(depth, maximumSearchDepth) && doNewIteration(stopCriterion, hasImproved, optimalLeaf.getCost(crac))) {
```
[the remark this answers is not in the document]
Comment (terriervik) [reply ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/process/search_tree/Tree.java:75]: Isn't it cleaner to separate them?

File: ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/process/search_tree/Tree.java (hunk 1 above)
Comment (terriervik) [reply ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/process/search_tree/Tree.java:108]: Can I put it protected, in order to keep the test I wrote in TreeTest.java?

File: ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/process/search_tree/Tree.java (hunk 2 above)
Comment (terriervik) [reply ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/process/search_tree/Tree.java:115]: same...

File: ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/process/search_tree/Tree.java (hunk 3 above)
Comment (terriervik) [reply ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/process/search_tree/Tree.java:122]: :penguin:

File: distribution/src/main/resources/config.yml [hunk 5]
```diff
@@ -70,3 +70,7 @@ linear-rao-parameters:
 
 search-tree-rao-parameters:
     range-action-rao : LinearRao
+    "stop-criterion" : POSITIVE_MARGIN
+    "maximum-search-depth" : 20
+    "relative-network-action-minimum-impact-threshold" : 0
+    "absolute-network-action-minimum-impact-threshold" : 0
```
Comment (baptsegu) [inline distribution/src/main/resources/config.yml:76]: i think the "" should not make it work

File: ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/process/search_tree/Tree.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (baptsegu) [reply ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/process/search_tree/Tree.java]: I don't know, both solutions are OK i guess...

**Pull request merged**